arrow/src/main/java/org/apache/iceberg/arrow/vectorized/VectorHolder.java

```diff
@@ -131,6 +131,20 @@ public boolean isDummy() {
     return vector == null;
   }
 
+  public static class NullVectorHolder extends VectorHolder {
```

Comment:
see my other comment on whether we should actually introduce this class vs modify `ConstantVectorHolder`to hold a `NullVector`